From 1148f8628b0cc4d64ffd168a65fc4d064af2e998 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Mon, 26 Feb 2018 17:41:35 +0200
Subject: [PATCH 0585/1051] dts: 8xxy: disable AP XOR engines on a8k-p devices

The CM3 firmware performs maintenance operation and uses
part of the XOR engines located on the AP.

The patch disables the required engines on A8082/4 and A8162/4
devices.

Change-Id: I0b3ea207dd8dd223cadc87eb13d7a4a8f607c1d0
Signed-off-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8082-db.dts |  8 ++++++++
 arch/arm64/boot/dts/marvell/armada-8084-db.dts | 16 ++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8162-db.dts |  8 ++++++++
 arch/arm64/boot/dts/marvell/armada-8164-db.dts | 16 ++++++++++++++++
 4 files changed, 48 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8082-db.dts b/arch/arm64/boot/dts/marvell/armada-8082-db.dts
index 61025b25763e..0566fe767b04 100644
--- a/arch/arm64/boot/dts/marvell/armada-8082-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8082-db.dts
@@ -61,6 +61,14 @@
 	};
 };
 
+&ap0_xor0 {
+	status = "disabled";
+};
+
+&ap0_xor1 {
+	status = "disabled";
+};
+
 &cp0_sata0 {
 	status = "okay";
 	sata-port@1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8084-db.dts b/arch/arm64/boot/dts/marvell/armada-8084-db.dts
index 047f3a85df33..e44fd3e91ede 100644
--- a/arch/arm64/boot/dts/marvell/armada-8084-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8084-db.dts
@@ -61,6 +61,22 @@
 	};
 };
 
+&ap0_xor0 {
+	status = "disabled";
+};
+
+&ap0_xor1 {
+	status = "disabled";
+};
+
+&ap0_xor2 {
+	status = "disabled";
+};
+
+&ap0_xor3 {
+	status = "disabled";
+};
+
 &cp0_sata0 {
 	status = "okay";
 	sata-port@1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8162-db.dts b/arch/arm64/boot/dts/marvell/armada-8162-db.dts
index 3559cdec6d4f..a33ff0b575d6 100644
--- a/arch/arm64/boot/dts/marvell/armada-8162-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8162-db.dts
@@ -61,6 +61,14 @@
 	};
 };
 
+&ap0_xor0 {
+	status = "disabled";
+};
+
+&ap1_xor0 {
+	status = "disabled";
+};
+
 &cp0_sata0 {
 	status = "okay";
 	sata-port@1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8164-db.dts b/arch/arm64/boot/dts/marvell/armada-8164-db.dts
index 7da5df65138f..d24f7f84e407 100644
--- a/arch/arm64/boot/dts/marvell/armada-8164-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8164-db.dts
@@ -61,6 +61,22 @@
 	};
 };
 
+&ap0_xor0 {
+	status = "disabled";
+};
+
+&ap0_xor1 {
+	status = "disabled";
+};
+
+&ap1_xor0 {
+	status = "disabled";
+};
+
+&ap1_xor1 {
+	status = "disabled";
+};
+
 &cp0_sata0 {
 	status = "okay";
 	sata-port@1 {
-- 
2.17.1

